### PR TITLE
Compute pathtracing PR

### PR DESCRIPTION
**Pull request by nipunG314**

### CODE REVIEW

File: 30_ComputeShaderPathTracer/include/nbl/this_example/transform.hpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+void EditTransform(float* cameraView, const float* cameraProjection, float* matrix, const TransformRequestParams& params)
+{
+	static bool useSnap = false;
+	static float snap[3] = { 1.f, 1.f, 1.f };
+	static float bounds[] = { -0.5f, -0.5f, -0.5f, 0.5f, 0.5f, 0.5f };
+	static float boundsSnap[] = { 0.1f, 0.1f, 0.1f };
+	static bool boundSizing = false;
+	static bool boundSizingSnap = false;
+
```
Comment (devshgraphicsprogramming) [inline 30_ComputeShaderPathTracer/include/nbl/this_example/transform.hpp:21]: why is this file here?

**Comment by devshgraphicsprogramming** [comment]
btw did you have to make any changes to the GLSL or was it a simple copy & paste from the previous  example?

**Comment by nipunG314** [comment]
> btw did you have to make any changes to the GLSL or was it a simple copy & paste from the previous example?

Copy and paste

File: 30_ComputeShaderPathTracer/include/nbl/this_example/transform.hpp (hunk 1 above)
Comment (nipunG314) [reply 30_ComputeShaderPathTracer/include/nbl/this_example/transform.hpp]: Removed

File: 30_ComputeShaderPathTracer/main.cpp [hunk 2]
```diff
@@ -730,6 +730,71 @@ class ComputeShaderPathtracer final : public examples::SimpleWindowedApplication
...
+						float freePercentage = 100.0f * (float)(totalAllocatedSize - isUse) / (float)totalAllocatedSize;
+						float allocatedPercentage = 1.0f - (float)(totalAllocatedSize - isUse) / (float)totalAllocatedSize;
+
+						ImVec2 barSize = ImVec2(400, 30);
+						float windowPadding = 10.0f;
+						float verticalPadding = ImGui::GetStyle().FramePadding.y;
+
+						ImGui::SetNextWindowSize(ImVec2(barSize.x + 2 * windowPadding, 110 + verticalPadding), ImGuiCond_Always);
+						ImGui::Begin("Nabla Imgui MDI Buffer Info", nullptr, ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoScrollbar);
+
```
Comment (devshgraphicsprogramming) [inline 30_ComputeShaderPathTracer/main.cpp:748]: you don;t need this debug, just allocate 1 MB you'll never have a problem

Comment (AnastaZIuk) [reply 30_ComputeShaderPathTracer/main.cpp:748]: just to clarify @nipunG314 , if you pull latest stuff you can allocate MDI streaming buffer [yourself](https://github.com/Devsh-Graphics-Programming/Nabla/blob/ee3fafe3dd03772da56743cf2a0a239fc483eef6/include/nbl/ext/ImGui/ImGui.h#L47) now, also make sure to have proper access, memory, allocation & usage flags by checking [this](https://github.com/Devsh-Graphics-Programming/Nabla/blob/ee3fafe3dd03772da56743cf2a0a239fc483eef6/src/nbl/ext/ImGui/ImGui.cpp#L675)

File: 30_ComputeShaderPathTracer/app_resources/present.frag [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2024-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#pragma wave shader_stage(fragment)
+
```
Comment (devshgraphicsprogramming) [inline 30_ComputeShaderPathTracer/app_resources/present.frag:6]: the reason this shader won't load as a bunch of HLSL is because of missing `.hlsl` extension, use `.frag.hlsl`

P.S. The `.frag` is redundant if you already have `#pragma wave shader_stage`

File: 30_ComputeShaderPathTracer/main.cpp [hunk 4]
```diff
@@ -34,7 +37,7 @@ class ComputeShaderPathtracer final : public examples::SimpleWindowedApplication
 		};
 
 		constexpr static inline uint32_t2 WindowDimensions = { 1280, 720 };
-		constexpr static inline uint32_t FramesInFlight = 5;
+		constexpr static inline uint32_t FramesInFlight = 1;
```
Comment (devshgraphicsprogramming) [inline 30_ComputeShaderPathTracer/main.cpp:40]: ekhm.... please use frames in flight

Comment (devshgraphicsprogramming) [reply 30_ComputeShaderPathTracer/main.cpp:40]: you can ask @Erfan-Ahmadi to proof read your code / give you pointers

File: 30_ComputeShaderPathTracer/app_resources/present.frag (hunk 3 above)
Comment (nipunG314) [reply 30_ComputeShaderPathTracer/app_resources/present.frag]: Done

File: 30_ComputeShaderPathTracer/main.cpp (hunk 2 above)
Comment (nipunG314) [reply 30_ComputeShaderPathTracer/main.cpp]: Done

File: 30_ComputeShaderPathTracer/main.cpp [hunk 5]
```diff
@@ -929,31 +905,26 @@ class ComputeShaderPathtracer final : public examples::SimpleWindowedApplication
...
+				SBasicViewParametersAligned pc;
+
+				SBasicViewParametersAligned viewParams;
+				memcpy(viewParams.params.MV, mv.pointer(), sizeof(mv));
+				memcpy(viewParams.params.MVP, mvp.pointer(), sizeof(mvp));
+				memcpy(viewParams.params.NormalMat, normalMat.pointer(), sizeof(normalMat));
+
+				memcpy(pc.params.MV, mv.pointer(), sizeof(mv));
+				memcpy(pc.params.MVP, mvp.pointer(), sizeof(mvp));
+				memcpy(pc.params.NormalMat, normalMat.pointer(), sizeof(normalMat));
```
Comment (devshgraphicsprogramming) [inline 30_ComputeShaderPathTracer/main.cpp:981]: only these two matrices need to be sent over

File: 30_ComputeShaderPathTracer/main.cpp [hunk 6]
```diff
@@ -986,8 +957,8 @@ class ComputeShaderPathtracer final : public examples::SimpleWindowedApplication
 				{
 					cmdbuf->bindComputePipeline(m_PTPipeline.get());
 					cmdbuf->bindDescriptorSets(EPBP_COMPUTE, m_PTPipeline->getLayout(), 0u, 1u, &m_descriptorSet0.get());
-					cmdbuf->bindDescriptorSets(EPBP_COMPUTE, m_PTPipeline->getLayout(), 1u, 1u, &m_uboDescriptorSet1.get());
 					cmdbuf->bindDescriptorSets(EPBP_COMPUTE, m_PTPipeline->getLayout(), 2u, 1u, &m_descriptorSet2.get());
+					cmdbuf->pushConstants(m_PTPipeline->getLayout(), IShader::E_SHADER_STAGE::ESS_COMPUTE, 0, sizeof(SBasicViewParametersAligned), &pc);
```
Comment (devshgraphicsprogramming) [inline 30_ComputeShaderPathTracer/main.cpp:1020]: use some other struct, `SBasicViewParametersAligned` are too big to fix in push constants

File: 30_ComputeShaderPathTracer/main.cpp (hunk 4 above)
Comment (nipunG314) [reply 30_ComputeShaderPathTracer/main.cpp]: Done

File: 30_ComputeShaderPathTracer/main.cpp (hunk 6 above)
Comment (nipunG314) [reply 30_ComputeShaderPathTracer/main.cpp]: Done

File: 30_ComputeShaderPathTracer/main.cpp (hunk 5 above)
Comment (nipunG314) [reply 30_ComputeShaderPathTracer/main.cpp]: Done

File: 30_ComputeShaderPathTracer/app_resources/common.glsl [hunk 7]
```diff
@@ -4,7 +4,6 @@
 
 // basic settings
 #define MAX_DEPTH 3
```
Comment (devshgraphicsprogramming) [inline 30_ComputeShaderPathTracer/app_resources/common.glsl:6]: you can also make the depth configurable in ImGUI

File: 30_ComputeShaderPathTracer/main.cpp [hunk 8]
```diff
@@ -1252,6 +1261,7 @@ class ComputeShaderPathtracer final : public examples::SimpleWindowedApplication
 		float camYAngle = 165.f / 180.f * 3.14159f;
 		float camXAngle = 32.f / 180.f * 3.14159f;
 		int PTPipline = E_LIGHT_GEOMETRY::ELG_SPHERE;
+		int spp = 128;
```
Comment (devshgraphicsprogramming) [inline 30_ComputeShaderPathTracer/main.cpp:1264]: good default is 16 or 32

Comment (nipunG314) [reply 30_ComputeShaderPathTracer/main.cpp]: Done

File: 30_ComputeShaderPathTracer/app_resources/common.glsl (hunk 7 above)
Comment (nipunG314) [reply 30_ComputeShaderPathTracer/app_resources/common.glsl]: Done
